src/pages/nutrition/productCard.jsx

```diff
@@ -0,0 +1,109 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import Zoom from "react-medium-image-zoom";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import offService from "../../off";
+
+// import { basicNutriments } from "./nutritionFields";
+
+export default function ProductNutriments({ setNutriments, nutriments }) {
+  const [page, setPage] = React.useState(1);
+  const [index, setIndex] = React.useState(0);
+  const [products, setProducts] = React.useState([]);
+
+  const product = products[index];
+  const productName = product
+    ? product.product_name
+      ? product.product_name
+      : "Brand Unknown"
+    : "Loading...";
+
+  React.useEffect(() => {
+    const productListUrl = offService.getNutritionToFillUrl({ page });
+
+    fetch(productListUrl)
+      .then((res) => res.json())
+      .then((data) => {
+        setProducts(data.products);
+      });
+  }, [page]);
+
+  const pictureURL = products[index]
+    ? products[index].image_nutrition_url
+    : "https://static.openfoodfacts.org/images/image-placeholder.png";
+
+  function clickHandler() {
+    setIndex((prev) => (prev < 22 ? ++prev : 0));
```

Comment:
To avoid looping on the same page

```suggestion
	if(index === products.length-1){
		// If you are on the last item, fetch a new page
		setPage(page+1)
		setIndex(0)
		return
	}
	setIndex((prev) => (prev < 22 ? ++prev : 0));	
```